### PR TITLE
Login: remove signin inheritance

### PR DESCRIPTION
**Pull request by nheagy**
Refs #7272

It's a good time to nuke the inheritance of `Signin*` classes for all new `Login*` classes.

To test:
- make sure logins work. Touched a lot here, tho hopefully my cut & paste skills were good.

Needs review: @aerych 
Or I can send it to someone more optional-friendly.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/LoginWPComViewController.swift [hunk 1]
```diff
@@ -76,22 +144,99 @@ class LoginWPComViewController: SigninWPComViewController, LoginViewController {
...
+    func handleKeyboardWillShow(_ notification: Foundation.Notification) {
+        keyboardWillShow(notification)
+    }
+
+    func handleKeyboardWillHide(_ notification: Foundation.Notification) {
+        keyboardWillHide(notification)
+    }
+
+    override func dismiss() {
+        self.performSegue(withIdentifier: .showEpilogue, sender: self)
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/LoginWPComViewController.swift]: Suggestion here.  Maybe rename this to `showEpilogue` or some such.  We might want to keep the original dismiss implementation intact in order to dismiss the controller if, for instance, a user who already has self-hosted blogs in the app decides to cancel the wpcom login flow.  Just a thought.

File: WordPress/Classes/ViewRelated/NUX/LoginWPComViewController.swift [hunk 2]
```diff
@@ -76,22 +144,99 @@ class LoginWPComViewController: SigninWPComViewController, LoginViewController {
...
+    }
+
+    func configureTextFields() {
+        usernameField?.text = loginFields.username
+        passwordField?.text = loginFields.password
+        passwordField?.textInsets = WPStyleGuide.edgeInsetForLoginTextFields()
+        emailLabel?.text = usernameField?.text
+    }
+
+    func localizeControls() {
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/LoginWPComViewController.swift]: Can we move localizeControls and configureTextFields into the configuration section above?

File: WordPress/Classes/ViewRelated/NUX/LoginWPComViewController.swift (hunk 1 above)
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/LoginWPComViewController.swift]: We can't make that change _yet_, as `dismiss` is part of the `SigninWPComSyncHandler` protocol.

**Comment by nheagy** [comment]
@aerych thanks for the feedback—all points taken :D 

Let me know what you think!

File: WordPress/Classes/ViewRelated/NUX/LoginLinkAuthViewController.swift [hunk 3]
```diff
@@ -1,14 +1,65 @@
...
+///
+class LoginLinkAuthViewController: NUXAbstractViewController, SigninWPComSyncHandler, LoginViewController {
+    @IBOutlet weak var statusLabel: UILabel?
+    var email: String = ""
+    var token: String = ""
+    var didSync: Bool = false
 
-class LoginLinkAuthViewController: SigninLinkAuthViewController, LoginViewController {
     // let the storyboard's style stay
     override func setupStyles() {}
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/LoginLinkAuthViewController.swift]: I think this one was overlooked

**Comment by nheagy** [comment]
@aerych I did `LoginLinkAuthViewController` as part of #7497 😀

**Pull request merged**